tests/background.py

```diff
@@ -361,7 +361,76 @@ def testCFHT_oldAPI(self):
         if display:
             ds9.mtv(mi, frame = 1)
 
-            
+    def testXY0(self):
+        """Test fitting the background to an image with nonzero xy0
+
+        The statsImage and background image should not vary with xy0
+        """
+        bgImageList = [] # list of background images, one per xy0
+        statsImageList = [] # list of stats images, one per xy0
+        for xy0 in (afwGeom.Point2I(0, 0), afwGeom.Point2I(1000, 500)):
+            afwdataDir = eups.productDir("afwdata")
+            if not afwdataDir:
+                print >> sys.stderr, "Skipping testCFHT as afwdata is not setup"
+                return
+
+            mi = afwImage.MaskedImageF(os.path.join(afwdataDir,
+                                                    "CFHT", "D4", "cal-53535-i-797722_1.fits"))
+            bbox = afwGeom.Box2I(afwGeom.Point2I(500, 2000), afwGeom.Point2I(2079, 4609))
+            mi = mi.Factory(mi, bbox, afwImage.LOCAL)
```

Comment:
If you are asking about line 380 "mi = mi.Factory(mi, bbox, afwImage.LOCAL)" all this is doing is extracting a smallish piece of the overall image. It's not the best way to do the job; I'll change the code to specify the bbox as part of the constructor. The non-zero XY0 is desirable for making sure Background honors xy0, but setting the xy0 directly does that, so it's slightly redundant in this case.